src/id.rs

```diff
@@ -25,12 +25,12 @@ use crate::{ArklibError, Result};
 )]
 pub struct ResourceId {
     pub data_size: u64,
-    pub crc32: u32,
+    pub blake3: [u8; 32],
 }
 
 impl Display for ResourceId {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "{}-{}", self.data_size, self.crc32)
+        write!(f, "{}-{:?}", self.data_size, self.blake3)
```

Comment:
Something like https://crates.io/crates/base64 should be used instead of `{:?}`.